### PR TITLE
create read-in module that knows about the ntuple format

### PR DESCRIPTION
**Pull request by jennyailin**
Given files with data (in this example: ttbar, qcd, and wjets)
Concatenates all events
Creates (number of events)x(number of related branches) np array
Under the names of:
jet_branches
muon_branches
photon_branches

### CODE REVIEW

File: readinformat.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+def root2pandas(files_path, tree_name, **kwargs):
+	files = glob.glob(files_path)
+	ss = stack_arrays([root2array(fpath, tree_name, **kwargs).view(np.recarray) for fpath in files])
+	try:
+        	return pd.DataFrame(ss)
+    	except Exception:
+        	return pd.DataFrame(ss.data)
+
+def x(events, phrase):
```
Comment (mickypaganini) [inline readinformat.py]: Please make the name of this function a bit more informative. In general, one letter names are reserved for ephemeral variables (indices, etc.).

File: readinformat.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	#slice related branches
+	all_events_jets = x(all_events, 'Jet')
+	all_events_muons = x(all_events, 'Muon')
+	all_events_photons = x(all_events, 'Photon')
+	all_events_weights = all_events['EventWeight']
+	all_events_y = all_events['y']
+	
+	
+	#convert to np arrays
+	jet_branches = all_events_jets.as_matrix()
```
Comment (mickypaganini) [inline readinformat.py]: you can save yourself three lines of code by adding `.as_matrix()` to what you return in your function `x()` above.

File: readinformat.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	muon_branches = all_events_muons.as_matrix()
+	photon_branches = all_events_jets.as_matrix()
+	w_branch = all_events_weights.values
```
Comment (mickypaganini) [inline readinformat.py]: similar comment as above: you can just add `.values` to line 47 and 48

File: readinformat.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+	y_branch = all_events_y.values
+	
+	return jet_branches, muon_branches, photon_branches, w_branch, y_branch
```
Comment (mickypaganini) [inline readinformat.py]: if you call something `blah_branches`, people in HEP will assume it's a list of names of branches, by convention. If you can, try to stick to the conventional naming strategies and call your matrices of features `X_jets` etc, the weights `w` and the targets `y`. Ideally you would want anybody that uses your code to immediately understand what it returns without having to go figure it out by reading your code.

File: readinformat.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+#test
+print readin('/home/ubuntu/jenny/ttbar.root', '/home/ubuntu/jenny/qcd.root','/home/ubuntu/jenny/ttbar.root')
```
Comment (mickypaganini) [inline readinformat.py]: No need for the test to be present in the module.

**Comment by mickypaganini** [comment]
Hey, I added inline comments to further improve your code. In addition, you should provide documentation to every one of your functions, by adding a docstring that provides a description of the function, a description of the arguments it expects and of the items it returns.

**Comment by jennyailin** [comment]
Thanks for the feedback, let me know if there are any further changes I should make!

**Comment by mickypaganini** [comment]
Fantastic! Thanks for implementing all the changes -- it looks great. One last thing: could you adjust your function `read_in(ttbar_path, qcd_path, wjets_path)` to match the signature `read_in(class_files_dict)` specified in `pipeline.py`? 
You should then use that dictionary (specifically, the number of its keys) to find out how many classes are present in the problem specified by the user; then look at the values corresponding to each key to get the list of files that belong to each class.

**Comment by mickypaganini** [comment]
This is incredible! Good job!

**Pull request merged**